modules/personalize_taxonomy_context/personalize_taxonomy_context.module

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/**
+ * @file personalize_taxonomy_context.module
+ * Provides a visitor context plugin for taxonomy.
+ */
+
+/**
+ * Implements hook_personalize_visitor_contexts().
+ */
+function personalize_taxonomy_context_personalize_visitor_context() {
+  $info = array();
+  $path = drupal_get_path('module', 'personalize_taxonomy_context') . '/plugins';
+  $info['taxonomy_context'] = array(
+    'path' => $path . '/visitor_context',
+    'handler' => array(
+      'file' => 'TaxonomyContext.inc',
+      'class' => 'TaxonomyContext',
+    ),
+  );
+  return $info;
+}
+
+/**
+ * This function is expected to exist and be used to retrieve terms.
+ *
+ * @param $nid
+ *   Node id.
+ * @param $vid
+ *   Vocabulary id.
+ * @return array
+ *   An array of string terms.
+ */
+function personalize_taxonomy_context_get_terms($nid, $vid) {
+  $query =
+    'SELECT {taxonomy_term_data}.name ' .
+    'FROM {taxonomy_term_data} ' .
+    'JOIN {taxonomy_vocabulary} ' .
+    'ON {taxonomy_term_data}.vid = {taxonomy_vocabulary}.vid ' .
+    'JOIN {taxonomy_index} ' .
+    'ON {taxonomy_term_data}.tid = {taxonomy_index.tid} ' .
+    'WHERE {taxonomy_index}.nid = :nid ' .
+    'AND {taxonomy_vocabulary}.vid = :vid';
+
```

Comment:
This function shouldn't actually be necessary. We should implement hook_node_view() and we'll have access to the taxonomy terms on the node. We need to then add the taxonomy term values to the js settings, like we do with user profile contexts. The difference is that with user profile, we can just do it in the getAssets() method as it's not reacting t viewing a particular piece of content: https://github.com/acquia/personalize_module/blob/7.x-1.x/plugins/visitor_context/UserProfileContext.inc#L85

For taxonomy terms, we have to hook into the node and add an array of settings to the #attached property somewhere. Or maybe we should hook into the viewing of each taxonomy term. I'm not sure which is best. I don't think we need to work too much about supporting entities other than nodes at this point though.